core/templates/pages/exploration-player-page/services/exploration-recommendations.service.spec.ts

```diff
@@ -0,0 +1,206 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for ExplorationRecommendationsService.
+ */
+
+
+import { async, TestBed } from '@angular/core/testing';
+import { ExplorationRecommendationsService } from
+  // eslint-disable-next-line max-len
+  'pages/exploration-player-page/services/exploration-recommendations.service';
+import { ContextService } from 'services/context.service';
+import { ServicesConstants } from 'services/services.constants';
+import { UrlService } from 'services/contextual/url.service';
+import { ExplorationRecommendationsBackendApiService } from
+  'domain/recommendations/exploration-recommendations-backend-api.service';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { LearnerExplorationSummary } from 'domain/summary/learner-exploration-summary.model';
```

Comment:
Import angular libs on top in a separate section then local imports. (separated by newline)
```suggestion
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { async, TestBed } from '@angular/core/testing';

import { ExplorationRecommendationsBackendApiService } from 'domain/recommendations/exploration-recommendations-backend-api.service';
import { LearnerExplorationSummary } from 'domain/summary/learner-exploration-summary.model';
import { ExplorationRecommendationsService } from 'pages/exploration-player-page/services/exploration-recommendations.service';
import { ContextService } from 'services/context.service';
import { UrlService } from 'services/contextual/url.service';
import { ServicesConstants } from 'services/services.constants';
```

Reply:
done